src/plugins/intel_cpu/src/nodes/subgraph.cpp

```diff
@@ -761,6 +761,7 @@ void Subgraph::prepareParams() {
 
     auto builder = [this, cache](const SubgraphKey& key) -> std::shared_ptr<SubgraphExecutor> {
         const auto& snippet = subgraph_attrs->snippet;
+        const auto scratchpad = getScratchPad();
```

Comment:
> If we want to use const ref here, the corresponding methods should return this type.

I've never heard about this rule. Why?

In my opinion this particular case is more about the intent declaration, rather than an actual extra copy elimination (since any decent compiler will produce optimal code here due to copy elision optimizations). When you copy an object by reference, you you usually need either to request some info from the object, or pass it further to other classes/functions (our case). If you (const) copy an object by value you assume that the object will change soon, and you need to save a particular state of the object before it does change.

In general, I follow Meyers advice (see [this](https://stackoverflow.com/a/270435) thread for a discussion): try to use references for non-build-in types where possible.